### ISSUE TITLE
[Feature request] Add Support for Row-Level Security (RLS) 

### ISSUE THREAD

**Issue by kevinvermeersch** [teableio/teable#450]
Like with Airtable and other alternatives I'm missing Row-Level Security (RLS) in Teable. If RLS is already a thing in the project, it'd be great to get some more detailed docs or examples on how to set it up right.

**Comment by tea-artist** [comment]
We're planning the related features, and we'll share the draft in discussion before we start.
Welcome to share some specific scenarios, it will be a great help to us

**Comment by kevinvermeersch** [comment]
@tea-artist: Thank you. I imagine it being used for a simple CRM solution where employees can only access their assigned contacts (rows).

**Comment by tea-artist** [comment]
Regarding Row-Level Security (RLS), it indeed is a critical feature for building applications like CRM solutions, especially in scenarios you mentioned where employees can only access the contact information assigned to them. Our strategy is as follows:

- **Open Source Version** is primarily aimed at developers and individual users, offering most of the basic functionalities. The goal here is to lower the barrier to entry and foster community growth.
- **Enterprise Version** includes advanced features such as enterprise management and advanced permission management, targeting enterprise users who need fine-grained permission control. 

Our approach is similar to some of the successful open-source projects in the market, like GitLab and Red Hat. They also adopt a similar strategy, providing a basic free version and an enterprise version with more advanced features. This model not only meets the needs of different user groups but also ensures the continuous development and innovation of the project.

We plan to introduce the upcoming Row-Level Security (RLS) feature in the Enterprise Version.

Regarding the general plan for the Enterprise Version:
1. **Enterprise-grade Image**: For enterprise users to deploy on their own. This option is suitable for businesses that want full control over their infrastructure.
2. **Officially Hosted SaaS Version**: Provides a ready-to-use service for enterprise users, including all advanced features.

This model allows us to offer flexible solutions to enterprise users with different needs, whether they prefer full control over their infrastructure or prioritize convenience and efficiency. At the same time, the open-source version remains a crucial foundation, ensuring free access to our basic functionalities.

**Comment by 1991jhf** [comment]
it will be nice to have configuration interface mapped to backend PostgreSQL RLS.

**Comment by KnudH** [comment]
I don't think, that RLS itself should be an enterprise feature, it's much more a security feature which will also prevent single users or small teams from accidentally modifying or deleting data. 

What you mention is, that this is also aimed for developers. A developer should have the ability to modify the RLS policies inside the database directly. So I think a compromise would be to map the current user (and roles) to the database so that developers can play around and work with RLS there. The real enterprise feature would be the user interface to modify RLS policies, in my opinion.

**Comment by tea-artist** [comment]
RLS implemented on the [Authority Matrix](https://help.teable.io/en/basic/authority-matrix) feature in EE
